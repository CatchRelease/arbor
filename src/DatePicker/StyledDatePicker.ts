import DayPicker from 'react-day-picker';
import styled from '@emotion/styled';
import { transparentize } from 'polished';
import 'react-day-picker/lib/style.css';

const StyledDatePicker = styled(DayPicker)`
  color: ${(props) => props.theme.colors.text.default};
  font-family: ${(props) => props.theme.brandFont};
  font-size: ${(props) => props.theme.fontSizes.size3};

  .DayPicker-Caption {
    font-size: ${(props) => props.theme.fontSizes.size4};
    font-weight: ${(props) => props.theme.fontWeights.regular};
    margin-bottom: ${(props) => props.theme.space.regular};
    padding-top: ${(props) => props.theme.space.smallest};
  }

  .DayPicker-Month {
    border-collapse: separate;
    border-spacing: 5px;
    margin: 0;
  }

  .DayPicker-Weekday {
    color: ${(props) => props.theme.colors.text.default};
    font-size: ${(props) => props.theme.fontSizes.size3};
  }

  .DayPicker-Day {
    border-radius: ${(props) => props.theme.radii.small};
    height: 28px;
    padding: 0;
    width: 28px;
  }

  .DayPicker-Day--today {
    background-color: ${(props) => props.theme.colors.background.default};
    color: ${(props) => props.theme.colors.text.dark};
    font-weight: ${(props) => props.theme.fontWeights.regular};
  }

  .DayPicker-Day--outside,
  .DayPicker-Day--disabled {
    color: ${(props) => transparentize(0.7, props.theme.colors.grey60)};
  }

  .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
    background-color: ${(props) => props.theme.colors.intent.brand.dark};
    color: ${(props) => props.theme.colors.intent.brand.light};
    font-weight: ${(props) => props.theme.fontWeights.regular};
  }

  &:not(.DayPicker--interactionDisabled)
    .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
    background-color: ${(props) => props.theme.colors.intent.brand.light};
    color: ${(props) => props.theme.colors.intent.brand.dark};
  }

  .DayPicker-NavBar {
    position: absolute;
    right: 0;
    top: 0;
  }
`;

export default StyledDatePicker;
