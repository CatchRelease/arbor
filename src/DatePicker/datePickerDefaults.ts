import DatePickerNavbar from './DatePickerNavbar';

const dayPickerDefaults = {
  navbarElement: DatePickerNavbar,
  showOutsideDays: true,
  weekdaysShort: ['S', 'M', 'T', 'W', 'T', 'F', 'S']
};

export default dayPickerDefaults;
