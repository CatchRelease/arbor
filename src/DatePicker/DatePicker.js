import React from 'react';

import datePickerDefaults from './datePickerDefaults';
import StyledDatePicker from './StyledDatePicker';

const DatePicker = props => (
  <StyledDatePicker {...datePickerDefaults} {...props} />
);

export default DatePicker;
