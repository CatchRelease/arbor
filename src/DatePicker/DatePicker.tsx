import { DayPickerProps } from 'react-day-picker';
import datePickerDefaults from './datePickerDefaults';
import StyledDatePicker from './StyledDatePicker';

const DatePicker = (props: DayPickerProps) => (
  <StyledDatePicker {...datePickerDefaults} {...props} />
);

export default DatePicker;
