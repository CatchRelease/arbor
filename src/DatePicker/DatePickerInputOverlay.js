import React from 'react';
import PropTypes from 'prop-types';

import Box from '../Box';
import DatePicker from './DatePicker';
import PopoverContent from '../Popover/PopoverContent';

const DatePickerInputOverlay = ({ classNames, children, ...props }) => (
  <Box position="absolute" zIndex="100">
    <PopoverContent className={classNames.overlayWrapper} {...props}>
      <DatePicker {...children.props} />
    </PopoverContent>
  </Box>
);

DatePickerInputOverlay.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  classNames: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default DatePickerInputOverlay;
