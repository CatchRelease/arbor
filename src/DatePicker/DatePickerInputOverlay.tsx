import { DayPickerProps } from 'react-day-picker';

import Box from '../Box';
import DatePicker from './DatePicker';
import PopoverContent, {
  Props as PopoverContentProps
} from '../Popover/PopoverContent';

type Props = PopoverContentProps & {
  children: { props: DayPickerProps };
  classNames: { overlayWrapper: string };
};

const DatePickerInputOverlay = ({ classNames, children, ...props }: Props) => (
  <Box position="absolute" zIndex={100}>
    <PopoverContent className={classNames.overlayWrapper} {...props}>
      <DatePicker {...children.props} />
    </PopoverContent>
  </Box>
);

export default DatePickerInputOverlay;
