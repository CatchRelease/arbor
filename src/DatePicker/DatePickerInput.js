import React from 'react';
import PropTypes from 'prop-types';
import DayPickerInput from 'react-day-picker/DayPickerInput';

import datePickerDefaults from './datePickerDefaults';
import DatePickerInputOverlay from './DatePickerInputOverlay';
import Input from '../Input/Input';

const DatePickerInput = ({ dayPickerProps, ...props }) => (
  <DayPickerInput
    component={Input}
    overlayComponent={DatePickerInputOverlay}
    dayPickerProps={{
      ...datePickerDefaults,
      ...dayPickerProps
    }}
    {...props}
  />
);

DatePickerInput.propTypes = {
  dayPickerProps: PropTypes.object // eslint-disable-line react/forbid-prop-types
};

DatePickerInput.defaultProps = {
  dayPickerProps: {}
};

export default DatePickerInput;
