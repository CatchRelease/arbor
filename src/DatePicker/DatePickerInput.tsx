import { FC } from 'react';
import { DayPickerInputProps } from 'react-day-picker';
import DayPickerInput from 'react-day-picker/DayPickerInput';

import datePickerDefaults from './datePickerDefaults';
import DatePickerInputOverlay from './DatePickerInputOverlay';
import Input, { InputProps } from '../Input';

const defaultInputProps = {
  autoComplete: 'off'
};

type Props = Omit<DayPickerInputProps, 'inputProps'> & {
  inputProps?: InputProps;
};

const DatePickerInput: FC<Props> = ({
  dayPickerProps = {},
  inputProps = {},
  ...props
}) => (
  <DayPickerInput
    component={Input}
    overlayComponent={DatePickerInputOverlay}
    dayPickerProps={{
      ...datePickerDefaults,
      ...dayPickerProps
    }}
    inputProps={{
      ...defaultInputProps,
      ...inputProps
    }}
    {...props}
  />
);

export default DatePickerInput;
