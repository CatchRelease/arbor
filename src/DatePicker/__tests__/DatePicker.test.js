import { shallow } from 'enzyme';

import createWithTheme from '../../../utils/createWithTheme';
import DatePicker from '../DatePicker';
import StyledDatePicker from '../StyledDatePicker';

// Explicitly set the value of today to a specific date so tests don't fail every time the day, month, or year changes
const today = new Date('2019-03-11T04:00:00.000Z');
const modifiers = { today };

describe('<DatePicker />', () => {
  it('properly renders a DatePicker', () => {
    const tree = createWithTheme(
      <DatePicker initialMonth={today} modifiers={modifiers} />
    );

    expect(tree).toMatchSnapshot();
  });

  describe('props', () => {
    it('passes props along to the StyledDatePicker', () => {
      const datePicker = shallow(
        <DatePicker modifiers={modifiers} initialMonth={today} foo="bar" />
      );

      expect(datePicker.find(StyledDatePicker)).toHaveProp({
        foo: 'bar'
      });
    });
  });
});
