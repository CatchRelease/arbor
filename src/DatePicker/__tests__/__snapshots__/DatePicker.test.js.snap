// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> properly renders a DatePicker 1`] = `
.emotion-14 {
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.8125rem;
}

.emotion-14 .DayPicker-Caption {
  font-size: 0.875rem;
  font-weight: 400;
  margin-bottom: 1rem;
  padding-top: 0.25rem;
}

.emotion-14 .DayPicker-Month {
  border-collapse: separate;
  border-spacing: 5px;
  margin: 0;
}

.emotion-14 .DayPicker-Weekday {
  color: #25272D;
  font-size: 0.8125rem;
}

.emotion-14 .DayPicker-Day {
  border-radius: 3px;
  height: 28px;
  padding: 0;
  width: 28px;
}

.emotion-14 .DayPicker-Day--today {
  background-color: #F2F2F2;
  color: #0F0F10;
  font-weight: 400;
}

.emotion-14 .DayPicker-Day--outside,
.emotion-14 .DayPicker-Day--disabled {
  color: rgba(152,154,156,0.3);
}

.emotion-14 .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
  background-color: #081D17;
  color: #C0F9DA;
  font-weight: 400;
}

.emotion-14:not(.DayPicker--interactionDisabled) .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
  background-color: #C0F9DA;
  color: #081D17;
}

.emotion-14 .DayPicker-NavBar {
  position: absolute;
  right: 0;
  top: 0;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 3px;
  font-family: Motiva Sans;
  font-weight: 500;
  line-height: 1;
  outline: none;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  padding: 0 0.25rem;
  height: 1.5rem;
  font-size: 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #5B5D61;
  border: 1px solid transparent;
}

.emotion-2:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-2:enabled {
  cursor: pointer;
}

.emotion-2 i {
  vertical-align: middle;
}

.emotion-2 i {
  color: #989A9C;
}

.emotion-2:hover:enabled {
  color: #0F0F10;
}

.emotion-2:hover:enabled i {
  color: #0F0F10;
}

.emotion-2:active:enabled {
  background-color: #F2F2F2;
  color: #0F0F10;
}

.emotion-2:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #081D17;
}

.emotion-2:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-0 {
  color: #989A9C;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-6 {
  box-sizing: border-box;
  margin-right: 0.75rem;
  margin-left: 0.75rem;
}

.emotion-4 {
  color: #989A9C;
  font-size: 1rem;
}

.emotion-8 {
  color: #989A9C;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

<div
  className="DayPicker DayPicker--interactionDisabled emotion-14 emotion-15"
  lang="en"
>
  <div
    className="DayPicker-wrapper"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="DayPicker-NavBar emotion-12 emotion-13"
    >
      <button
        aria-label="Previous Month"
        className="button--minimal emotion-2 emotion-3"
        disabled={false}
        onClick={[Function]}
        size="small"
      >
        <i
          className="ar ar-chevron emotion-0 emotion-1"
          color="icon.default"
        />
      </button>
      <div
        className="emotion-6 emotion-7"
      >
        <i
          className="ar ar-symbol-circle emotion-4 emotion-1"
          color="icon.default"
          fontSize="size5"
        />
      </div>
      <button
        aria-label="Next Month"
        className="button--minimal emotion-2 emotion-3"
        disabled={false}
        onClick={[Function]}
        size="small"
      >
        <i
          className="ar ar-chevron emotion-8 emotion-1"
          color="icon.default"
        />
      </button>
    </div>
    <div
      className="DayPicker-Months"
    >
      <div
        className="DayPicker-Month"
        role="grid"
      >
        <div
          className="DayPicker-Caption"
          role="heading"
        >
          <div
            onKeyUp={[Function]}
          >
            March 2019
          </div>
        </div>
        <div
          className="DayPicker-Weekdays"
          role="rowgroup"
        >
          <div
            className="DayPicker-WeekdaysRow"
            role="row"
          >
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Sunday"
              >
                S
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Monday"
              >
                M
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Tuesday"
              >
                T
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Wednesday"
              >
                W
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Thursday"
              >
                T
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Friday"
              >
                F
              </abbr>
            </div>
            <div
              className="DayPicker-Weekday"
              role="columnheader"
            >
              <abbr
                title="Saturday"
              >
                S
              </abbr>
            </div>
          </div>
        </div>
        <div
          className="DayPicker-Body"
          role="rowgroup"
        >
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={true}
              aria-label="Sun Feb 24 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              24
            </div>
            <div
              aria-disabled={true}
              aria-label="Mon Feb 25 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              25
            </div>
            <div
              aria-disabled={true}
              aria-label="Tue Feb 26 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              26
            </div>
            <div
              aria-disabled={true}
              aria-label="Wed Feb 27 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              27
            </div>
            <div
              aria-disabled={true}
              aria-label="Thu Feb 28 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              28
            </div>
            <div
              aria-disabled={false}
              aria-label="Fri Mar 01 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              1
            </div>
            <div
              aria-disabled={false}
              aria-label="Sat Mar 02 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              2
            </div>
          </div>
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={false}
              aria-label="Sun Mar 03 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              3
            </div>
            <div
              aria-disabled={false}
              aria-label="Mon Mar 04 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              4
            </div>
            <div
              aria-disabled={false}
              aria-label="Tue Mar 05 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              5
            </div>
            <div
              aria-disabled={false}
              aria-label="Wed Mar 06 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              6
            </div>
            <div
              aria-disabled={false}
              aria-label="Thu Mar 07 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              7
            </div>
            <div
              aria-disabled={false}
              aria-label="Fri Mar 08 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              8
            </div>
            <div
              aria-disabled={false}
              aria-label="Sat Mar 09 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              9
            </div>
          </div>
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={false}
              aria-label="Sun Mar 10 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              10
            </div>
            <div
              aria-disabled={false}
              aria-label="Mon Mar 11 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--today"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              11
            </div>
            <div
              aria-disabled={false}
              aria-label="Tue Mar 12 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              12
            </div>
            <div
              aria-disabled={false}
              aria-label="Wed Mar 13 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              13
            </div>
            <div
              aria-disabled={false}
              aria-label="Thu Mar 14 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              14
            </div>
            <div
              aria-disabled={false}
              aria-label="Fri Mar 15 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              15
            </div>
            <div
              aria-disabled={false}
              aria-label="Sat Mar 16 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              16
            </div>
          </div>
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={false}
              aria-label="Sun Mar 17 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              17
            </div>
            <div
              aria-disabled={false}
              aria-label="Mon Mar 18 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              18
            </div>
            <div
              aria-disabled={false}
              aria-label="Tue Mar 19 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              19
            </div>
            <div
              aria-disabled={false}
              aria-label="Wed Mar 20 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              20
            </div>
            <div
              aria-disabled={false}
              aria-label="Thu Mar 21 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              21
            </div>
            <div
              aria-disabled={false}
              aria-label="Fri Mar 22 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              22
            </div>
            <div
              aria-disabled={false}
              aria-label="Sat Mar 23 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              23
            </div>
          </div>
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={false}
              aria-label="Sun Mar 24 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              24
            </div>
            <div
              aria-disabled={false}
              aria-label="Mon Mar 25 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              25
            </div>
            <div
              aria-disabled={false}
              aria-label="Tue Mar 26 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              26
            </div>
            <div
              aria-disabled={false}
              aria-label="Wed Mar 27 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              27
            </div>
            <div
              aria-disabled={false}
              aria-label="Thu Mar 28 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              28
            </div>
            <div
              aria-disabled={false}
              aria-label="Fri Mar 29 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              29
            </div>
            <div
              aria-disabled={false}
              aria-label="Sat Mar 30 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              30
            </div>
          </div>
          <div
            className="DayPicker-Week"
            role="row"
          >
            <div
              aria-disabled={false}
              aria-label="Sun Mar 31 2019"
              aria-selected={false}
              className="DayPicker-Day"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              31
            </div>
            <div
              aria-disabled={true}
              aria-label="Mon Apr 01 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              1
            </div>
            <div
              aria-disabled={true}
              aria-label="Tue Apr 02 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              2
            </div>
            <div
              aria-disabled={true}
              aria-label="Wed Apr 03 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              3
            </div>
            <div
              aria-disabled={true}
              aria-label="Thu Apr 04 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              4
            </div>
            <div
              aria-disabled={true}
              aria-label="Fri Apr 05 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              5
            </div>
            <div
              aria-disabled={true}
              aria-label="Sat Apr 06 2019"
              aria-selected={false}
              className="DayPicker-Day DayPicker-Day--outside"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="gridcell"
              tabIndex={-1}
            >
              6
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
